src/checkpoint/traits/serializable_traits.h

```diff
@@ -171,9 +176,12 @@ struct SerializableTraits {
   static constexpr auto const has_serialize_function =
     has_serialize_instrusive or has_serialize_noninstrusive;
 
+  static constexpr auto const is_constructible =
+    is_default_constructible or is_reconstructible or is_tagged_constructible;
+
   // This defines what it means to be serializable
   static constexpr auto const is_serializable =
-    has_serialize_function and (is_default_constructible or is_reconstructible);
+    has_serialize_function and is_constructible;
```

Comment:
Could you order the definitions in this class to group them by 'category' of use? I.e. byte copy and arithmetic, then construction, then serialization, then composites that mix those?

I'm fine with having all the `constexpr` values after all the type definitions, as long as the grouping is applied within each. It's just jarring to see everything kinda tossed together.

Reply:
Ok, just pushed a whole re-order and cleanup of traits.